### PR TITLE
FEATURE | Added support for anonymous classes

### PR DESCRIPTION
**Pull request by JustBlackBird**

### CODE REVIEW

File: src/Parser/Visitor/ExtractDeclaredNamespace.php [hunk 1]
```diff
@@ -12,7 +12,7 @@ class ExtractDeclaredNamespace extends NodeVisitorAbstract
 
     public function enterNode(Node $node)
     {
-        if ($node instanceof Node\Stmt\ClassLike && $node->namespacedName !== null) {
+        if ($node instanceof Node\Stmt\ClassLike && isset($node->namespacedName) && $node->namespacedName  !== null) {
```
Comment (JustBlackBird) [inline src/Parser/Visitor/ExtractDeclaredNamespace.php:15]: Anonymous classes has no `namespacedName` field. So comparing it with null ends up with "Undefined field" notice

**Comment by j6s** [comment]
Just to be clear on what this does: What exactly are you referring to with anonymous classes?

**Comment by JustBlackBird** [comment]
There are [classes that do not have names](http://php.net/manual/en/language.oop5.anonymous.php) :)

Here is an example

```php
class Foo {
   public function createChild()
   {
       // This is an anonymous class
       return new class {
           public function doSomethig()
           {
           }
       }
   }
}
```

**Comment by j6s** [comment]
Looking at the test code I see what the problem is. To be honest, I had never seen anonymous classes in PHP.

You might want to have a look at the travis results:
PHPArch has a very broad dependency definition by design (In order to avoid dependency clashes with other static analysis tools that use the php-parser in earlier versions) - it seems like this change is incompatible with earlier versions of one of the dependencies.

**Comment by JustBlackBird** [comment]
I've seen Travis results. Working on the fix :)

File: src/Parser/Visitor/DocBlockTypeAnnotations.php [hunk 2]
```diff
@@ -8,21 +8,24 @@
...
 
     public function enterNode(Node $node)
     {
-        if ($node instanceof Node\Stmt\ClassLike && $node->namespacedName !== null) {
-            $this->lastClass = $node->namespacedName->toString();
-        } elseif ($node instanceof Node\Stmt\ClassMethod && $this->lastClass) {
-            $contextFactory = new \phpDocumentor\Reflection\Types\ContextFactory();
-            $context = $contextFactory->createFromReflector(new ReflectionMethod($this->lastClass, (string) $node->name));
+        if ($node instanceof Node\Stmt\Namespace_) {
+            $this->lastNamespace = $node->name->toString();
```
Comment (j6s) [inline src/Parser/Visitor/DocBlockTypeAnnotations.php]: Heads up: `$node->name` can either be `PhpParser\Node\Name` or `null` - please add a null check before calling `toString`.

Comment (JustBlackBird) [reply src/Parser/Visitor/DocBlockTypeAnnotations.php]: Fixed

**Comment by j6s** [comment]
Looks good, thank you for your contribution.

**Pull request merged**